agent/src/softwarecontaineragent.h

```diff
@@ -47,6 +48,66 @@ namespace softwarecontainer {
 
 static constexpr ContainerID INVALID_CONTAINER_ID = -1;
 
+class SoftwareContainerAgentError : public SoftwareContainerError
+{
+public:
+    SoftwareContainerAgentError():
+        m_message("SoftwareContainer error")
+    {
+    }
+
+    SoftwareContainerAgentError(const std::string &message):
+        m_message(message)
+    {
+    }
+
+    virtual const char *what() const throw()
+    {
+        return m_message.c_str();
+    }
+
+protected:
+    std::string m_message;
+};
+
+class PreloadError : public SoftwareContainerAgentError
+{
+public:
+    PreloadError():
+        SoftwareContainerAgentError("Software Container error - Failed to preload container")
+    {
+    }
+
+    PreloadError(const std::string &message):
+        SoftwareContainerAgentError(message)
+    {
+    }
+
+    virtual const char *what() const throw()
+    {
+        return m_message.c_str();
+    }
+};
+
+class WorkspaceError : public SoftwareContainerAgentError
```

Comment:
This error is not used anywhere yet. Also, to me it makes little sense to have this inherit the agent errors, since the workspace is part of the library.